test/bats/tests/fileformat/nginx-pod-secrets-store-inline-volume-java.yaml

```diff
@@ -0,0 +1,45 @@
+kind: Pod
+apiVersion: v1
+metadata:
+  name: nginx-secrets-store-inline-java
+spec:
+  containers:
+    - image: $CONTAINER_IMAGE
+      name: nginx
+      volumeMounts:
+        - name: secrets-store-inline
+          mountPath: "/mnt/secrets-store"
+          readOnly: true
+  volumes:
+    - name: secrets-store-inline
+      csi:
+        driver: secrets-store.csi.k8s.io
+        readOnly: true
+        volumeAttributes:
+          providerName: "azure"
+          usePodIdentity: "false"         # [OPTIONAL] if not provided, will default to "false"
+          keyvaultName: "$KEYVAULT_NAME" # the name of the KeyVault
+          fileFormatting: "JavaProperties"
```

Comment:
Should the `fileFormatting` be defined in the objects array? It's possible the file format is different for different objects in the same `SecretProviderClass`?

Also, the volume attributes are defined in `SecretProviderClass` [example](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/examples/v1alpha1_secretproviderclass.yaml) and referenced in pods  [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/examples/nginx-pod-secrets-store-inline-volume-secretproviderclass-podid.yaml#L20-L21). The option to define the provider specific parameters in pod spec was deprecated a while back and will be removed in the next release of the driver

Reply:
Hi @aramase ! Thanks for your comments, as I am definitely willing to improve the implementation.

I think the implementation should not be in the provider specific part, you are right. I will try to improve that part, but it might be harder as providers then all should (try to) support it. What is the right place to discuss these kind of design questions, is there a Slack channel and/or meeting I can join?

With regard to the formatting for the different objects, in my mind it would be confusing to have multiple formatting parameters, especially because the new fileformatting options are combining the objects. It can be done, but it feels a bit like overengineering the feature at this moment in time. It is not that hard to add an extra SecretProviderClass if two formattings are required. WDYT?

Reply:
The `spec.parameters` in `SecretProviderClass` is specific to each provider - https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/examples/v1alpha1_secretproviderclass.yaml#L7-L23. `spec.parameters` is a map, so it's extensible for each provider. 

> What is the right place to discuss these kind of design questions, is there a Slack channel and/or meeting I can join?

For driver, we have a slack channel and community meetings - https://github.com/kubernetes-sigs/secrets-store-csi-driver#want-to-help. It would be great if you want to join next week and discuss any changes or questions you have.

> It can be done, but it feels a bit like overengineering the feature at this moment in time. It is not that hard to add an extra SecretProviderClass if two formattings are required. WDYT?

Makes sense!